cmake/oneDPLWindowsIntelLLVMApply.cmake

```diff
@@ -36,6 +36,6 @@ endif()
 
 if (INTELLLVM_WIN_STD_IGNORE_FIX)
     include(Compiler/CMakeCommonCompilerMacros)
-    # Setting c++ standard default to 14, and enabling usage of CMAKE_CXX_STANDARD
+    # For IntelLLVM versions greater or equal to 2020, setting c++ standard default to 14, and enable usage of CMAKE_CXX_STANDARD
     __compiler_check_default_language_standard(CXX 2020 14)
```

Comment:
Should the default be set to C++17 now?

Reply:
I suppose the correct thing would be to make it so that version 2020 defaults to 14, and 2021 defaults to 17.  I think this is when icpx switched to c++17 default,  I can check what current cmake intelLLVM code does here if we want to update it to be a more full "fix".    

However, this is only as a workaround for windows cmake in windows, so it is not really important.  Within oneDPL's cmake (after this applied workaround), we set the default to c++17 if left unspecified.  The important part of this call is the other effect, which is to enable the use of `CMAKE_CXX_STANDARD` to specify a standard library version.

Reply:
I think actually this is still what CMake still calls for intelLLVM 

https://github.com/Kitware/CMake/blob/8b90d19630b0a80eed34df340f45eb1d61aff0ca/Modules/Compiler/IntelLLVM-CXX.cmake#L103

I'd be inclined to leave this as is, especially since this default value really goes unused in practice.

Reply:
Thanks for clarifying comment. It clearly says what the call does. I think the most important comment would be also the reason why is it needed, which you explained here. 

Something like this: _"This is a workaround for Windows cmake. This function side effect is that it enables the use of CMAKE_CXX_STANDARD to specify a standard library version. Otherwise CMAKE_CXX_STANDARD is ignored."_